File: Patterns/C#/Creational/Builder/BuilderJacketBeret.cs
public class BuilderJacketBeret : Builder{

    public BuilderJacketBeret(){
        outfit = new Outfit();
    }
    public override void buildCoat(){
        outfit.addClothe( new Jacket() );
    }

    public override void buildHat(){
        outfit.addClothe( new Beret() );
    }

    public override Outfit getOutfit(){
        return outfit;
    }

}

File: Patterns/Java/Creational/Builder/BuilderJacketBeret.java
public class BuilderJacketBeret extends Builder{

    BuilderJacketBeret(){
        outfit = new Outfit();
    }

    public void buildCoat(){
       outfit.addClothe( new Jacket() );
    }


    public void buildHat(){
        outfit.addClothe( new Beret() );
    }

    public Outfit getOutfit(){
        return outfit;
    }

}
